### PR TITLE
Reduce size of source constructor symbol

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
I didn't do this before because constructors still had data in them about if they were expression bodied or not.  But we moved that into the flags.  So we can trim this as well, removing all fields from this type.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@RikkiGibson @jcouv ptal.

**Review by 333fred — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs [hunk 1]
```diff
@@ -387,9 +400,10 @@ protected void CheckFileTypeUsage(TypeWithAnnotations returnType, ImmutableArray
             bool isExtensionMethod,
             bool isNullableAnalysisEnabled,
             bool isVarArg,
-            bool isExplicitInterfaceImplementation)
+            bool isExplicitInterfaceImplementation,
+            bool hasThisInitializer = false)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs:404]: >bool hasThisInitializer = false) [](http://example.com/codeflow?start=12&length=32)

Optional parameters are bug prone. It is easy to fail to pass value for them. I understand that `this` initializers are not common, but I think it is better to have no optional parameters in this API.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs:404]: @AlekseyTs sgtm.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Pull request merged**